lib/recipes-data/src/lib/extract-recipes.ts

```diff
@@ -0,0 +1,37 @@
+import type {Block} from "@guardian/content-api-models/v1/block";
+import type {Blocks} from "@guardian/content-api-models/v1/blocks";
+import type {Content} from "@guardian/content-api-models/v1/content";
+import {ContentType} from "@guardian/content-api-models/v1/contentType";
+import {ElementType} from "@guardian/content-api-models/v1/elementType";
+import type {RecipeReferenceWithoutChecksum} from './models';
+
+export function extractAllRecipesFromArticle(content: Content): Array<RecipeReferenceWithoutChecksum | null> {
+  if (content.type == ContentType.ARTICLE && content.blocks) {
+    const articleBlocks: Blocks = content.blocks
+    const getAllMainBlockRecipesIfPresent: Array<RecipeReferenceWithoutChecksum | null> = extractRecipeData(content.id, articleBlocks.main as Block)
+    const bodyBlocks = articleBlocks.body as Block[]
+    const getAllBodyBlocksRecipesIfPresent: Array<RecipeReferenceWithoutChecksum | null> = bodyBlocks.flatMap(bodyBlock => extractRecipeData(content.id, bodyBlock))
+    return getAllMainBlockRecipesIfPresent.concat(getAllBodyBlocksRecipesIfPresent)
```

Comment:
would you mind adding on a `.filter(recep=>!!recep)` here and then changing the return type back to `RecipeReferenceWithoutChecksum[]` ? Would make it simpler from calling code